### PR TITLE
Digital Garage RTD module: new rtd submodule #6410

### PR DESCRIPTION
**Pull request by mediaconsortium-develop**
## Type of change

- [ X ] Feature
https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ X ] Other

## Description of change

We want to add Real Time Data Submodule

- test parameters for validating bids
```
{
  bidder: '<bidder name>',
  params: {
    // ...
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.
https://www.suuiiisya.com/test/agytest/rt_module_test.html

- contact email of the adapter’s maintainer
mediaconsortium-develop@bi.garage.co.jp

- [ ] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- A link to a PR on the docs repo at https://github.com/prebid/prebid.github.io/
https://github.com/mediaconsortium-develop/Prebid.js/blob/feature/dgkeywordRtdProvider/modules/dgkeywordRtdProvider.md

## Other information

we want to set some keywords before requestBids() in our customer's site.
The keywords are real time data and can be got to access our partner's service site.
So we want to start requestBids() after getting keywords only when our keywords service use.

### CODE REVIEW

File: modules/dgkeywordRtdProvider.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
+## Overview
+
+Module Name: Digital Garage Keyword Module  
+Module Type: Rtd Provider  
+Maintainer: mediaconsortium-develop@bi.garage.co.jp  
+Remarks: This module can work only with AppNexusBidAdapter.
```
Comment (patmmccann) [inline modules/dgkeywordRtdProvider.md:6]: Please delete this line

**Comment by mediaconsortium-develop** [comment]
I delete Overview.
And push to remote.

2021年5月27日(木) 19:45 'Patrick McCann' via
***@***.***
***@***.***>:

**Comment by osazos** [comment]
It seems you have issue with tests themselves…

```Error: Uncaught AssertionError: expected { Object (opeaud, opectx) } to be an undefined (node_modules/chai/chai.js:239)```

**Comment by patmmccann** [comment]
I saw this on another adapter, tests passed this now

**Comment by osazos** [comment]
@mediaconsortium-develop, the tests were green but I'm pretty sure it was a false-positif… and after merging, they still fail  Can you check again why your tests don't pass the CI?

Thx

**Comment by mediaconsortium-develop** [comment]
I understood(guess) why ci test error occurred in another modules test.
our module set keywords in bidderConfig using setBidderConfig().
but in the same time, another module ci test is running and error occurs
because of our change of bidderConfig.

to avoid ci error, I tried the followings but can not stop ci error.
・when test code is executed, resetConfig() or setBidderConfig({ bidders:
[bidders], { ortb2: {} } }) using beforeEach(), afterEach() after()
・I make new function named resetBidderConfig() in config.js and execute
resetBidderConfig()  using beforeEach(), afterEach() after()
・when setBidderConfig() is executed,  exusecute resetConfig or
resetBidderConfig() in the test code as soon as possible(next line of it())

Now I thinked If realTimeData.dataProvider execute some setBidderrConfig,
some ci error may occurs in another  bidder driver test.
the error is like this.
==================
  ✗ "before each" hook
AssertionError: expected { Object (xxxxxx, ・・・) } to be an undefined
(node_modules/chai/chai.js:243)
===================

So, I changed the test code and module.
・we can select to execute setBidderConfig() or not. ※default: exetute
setBidderConfig
・in test code, set not to execute setBidderConfig because ci error occurs.

you can easily change to execute setBidderConfig in test code to change the
following setup.
========================
dgkeywordRtdProvider_spec.js
line: 7
const IGNORE_SET_ORTB2 = false;
========================

this is the new PR.
https://github.com/prebid/Prebid.js/pull/6912

By the way,
ci test have another error. but the error is the timeout of http request,
this error sometimes occurs because of the delay of network or server
response.
====================
timeout error.
   Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure
"done()" is called; if returning a Promise, ensure it resolves.
====================


2021年6月1日(火) 2:53 'Olivier' via ***@***.*** <
***@***.***>:

**Pull request merged**